cmd/notary/tuf.go

```diff
@@ -385,17 +386,35 @@ func (t *tufCommander) tufVerify(cmd *cobra.Command, args []string) error {
 		return fmt.Errorf("error retrieving target by name:%s, error:%v", targetName, err)
 	}
 
-	// Create hasher and hash data
-	stdinHash := sha256.Sum256(payload)
-	serverHash := target.Hashes["sha256"]
-
-	if subtle.ConstantTimeCompare(stdinHash[:], serverHash) == 0 {
-		return fmt.Errorf("notary: data not present in the trusted collection")
+	if err := checkHash(payload, target.Hashes); err != nil {
+		return err
 	}
+
 	_, _ = os.Stdout.Write(payload)
 	return nil
 }
 
+func checkHash(payload []byte, hashes data.Hashes) error {
+	// k, v indicate the hash algorithm and the corresponding value
+	for k, v := range hashes {
+		switch k {
+		case "sha256":
```

Comment:
nit: it'd be great if "sha256" and "sha512" could be constants in `const.go` (we could refactor `NewFileMeta` too)